pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -333,6 +355,121 @@ func (p *PoolManager) revertAllocationOnVm(vmName string, allocations map[string
 	p.releaseMacAddressesFromInterfaceMap(allocations)
 }
 
+func (p *PoolManager) getOrCreateVmMacWaitMap() (map[string]string, error) {
```

Comment:
do you think it would make sense to keep these helpers in a separate module. this logic is not trivial and it would be easier to document there

Reply:
I will prefer to document it here as we are going to have the same functions for pods in the future.

I will add comments onto of every function